samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -265,15 +268,22 @@ protected Object getEnvelopePartitionId(OutgoingMessageEnvelope envelope) {
   }
 
   private String convertPartitionKeyToString(Object partitionKey) {
+    String ret;
     if (partitionKey instanceof String) {
-      return (String) partitionKey;
+      ret = (String) partitionKey;
     } else if (partitionKey instanceof Integer) {
-      return String.valueOf(partitionKey);
+      ret = String.valueOf(partitionKey);
     } else if (partitionKey instanceof byte[]) {
-      return new String((byte[]) partitionKey, Charset.defaultCharset());
+      ret = new String((byte[]) partitionKey, Charset.defaultCharset());
     } else {
       throw new SamzaException("Unsupported key type: " + partitionKey.getClass().toString());
     }
+    if (ret != null && ret.length() > ClientConstants.MAX_PARTITION_KEY_LENGTH) {
+      LOG.debug("length of partitioning key ({}) exceeds limit: {}. Will truncate the key...", ret.length(),
+          ClientConstants.MAX_PARTITION_KEY_LENGTH);
+      ret = ret.substring(0, ClientConstants.MAX_PARTITION_KEY_LENGTH);
+    }
```

Comment:
Add documentation to this effect in the https://samza.apache.org/learn/documentation/0.14/azure/eventhubs.html